### PR TITLE
Implement addControls feature with graph/point control modes and input validation

### PR DESCRIPTION
**Pull request by dqnykamp**
Generalize point slider controls into an `addControls` feature with graph-level rendering modes. Replace the boolean `addSliders` attribute with a text `addControls` attribute supporting `all`, `slidersOnly`, `inputsOnly`, and `none`.

### Worker Components
- Rename `addSliders` → `addControls` in `Graph.js` and `Point.js`
- Change `Graph.addControls` to text type with valid values and fallback mappings (`true` → `"all"`, `false` → `"none"`)
- Rename `draggablePointsForSliders` → `draggablePointsForControls` state variable

### Renderer (`prefigure.tsx`)
- Implement control mode normalization (`normalizeGraphControlsMode`, `normalizePointControlsMode`) and active-input computation
- Add input parsing for single values (`parseSingleMathNumber`) and ordered pairs (`parseOrderedPair`)
- Implement input submission handlers (`submitAxisInput`, `submitPairInput`) with validation and error state management
- Guards: inputs only commit when a draft exists (blur on an untouched field is a no-op); no-op submits are skipped when the parsed value matches the current coordinate; `committingInputKeysRef` in-flight lock prevents Enter+blur duplicate dispatches
- Add inline axis inputs in `all` mode within slider labels
- Render pure input controls in `inputsOnly` mode
- Add error message display with proper ARIA associations (`aria-invalid`, `aria-describedby`)
- Fix layout jitter on invalid input by constraining label/card widths

### Accessibility & UX
- `aria-invalid` and `aria-describedby` for invalid inputs
- Per-point control card rendered as `role="group"` labelled by the point name heading (`aria-labelledby`)
- Outer controls section rendered as `role="group"` with `aria-label="Point controls"`
- Visually-hidden `aria-live="polite"` region for build/error status announcements, placed outside `role="img"` so announcements are not suppressed
- Graph `aria-label` defaults to `"Diagram"` when no `shortDescription` is provided
- Maintain DOM order for keyboard/screen-reader navigation
- Block `movePoint` on invalid input; show validation errors instead

### Tests
- Migrate `prefigureSliders.cy.js` tests to `addControls` semantics
- Add tests for `all`/`slidersOnly`/`inputsOnly`/`none` modes and validation behavior
- Add tests for position fallback and responsive layout

### Schema
- Regenerate schema artifacts with `addControls` attribute mappings and autocomplete values

### Maintenance
- Extract shared record pruning logic into `pruneRecordByActiveKeys` helper
- Hoist draft-update logic into `setInputDraftValue` helper

## FEATURE MODES

**`slidersOnly`**: Traditional slider interaction (existing behavior, renamed from `addSliders`)

**`inputsOnly`**: Text input controls where users enter single values or ordered pairs, validated as math expressions before committing

**`all`**: Sliders paired with editable inline axis inputs in slider labels; users can drag, type, or both

**`none`**: No controls rendered (default)

## TESTING
All existing Cypress tests pass. New tests cover all control modes, validation behavior, and responsive layout fallback.

### CODE REVIEW

File: .changeset/add-controls-generalized.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+"@doenet/doenetml": patch
+"@doenet/standalone": patch
+"@doenet/doenetml-iframe": patch
+---
+
+Generalize point slider controls into an `addControls` feature with graph-level modes. Replace the boolean `addSliders` attribute with `addControls` text attribute supporting `all`, `slidersOnly`, `inputsOnly`, and `none`. Rename the per-point `addSliders` attribute to `addControls` while preserving point-level options (`both`, `xOnly`, `yOnly`, `none`).
+
```
[the remark this answers is not in the document]
Comment (dqnykamp) [reply .changeset/add-controls-generalized.md:8]: breaking convention and ignoring this since old attribute was just added yesterday.

File: packages/doenetml/src/Viewer/renderers/utils/SliderUI.tsx [hunk 2]
```diff
@@ -36,7 +36,7 @@ const RANGE_INPUT_KEYS = new Set([
  */
 type SliderUIProps = {
     id: string;
-    label: string;
+    label: React.ReactNode;
     ariaLabel: string;
```
[the remark this answers is not in the document]
Comment (dqnykamp) [reply packages/doenetml/src/Viewer/renderers/utils/SliderUI.tsx:40]: fixed

File: packages/doenetml/src/Viewer/renderers/utils/SliderUI.tsx [hunk 3]
```diff
@@ -102,8 +102,18 @@ export default function SliderUI({
...
+                htmlFor={id}
+                style={{
+                    display: "block",
+                    width: "100%",
+                    minWidth: 0,
+                    overflowWrap: "anywhere",
+                }}
+            >
+                {label}
+            </label>
```
[the remark this answers is not in the document]
Comment (dqnykamp) [reply packages/doenetml/src/Viewer/renderers/utils/SliderUI.tsx]: fixed
